lsp/server/src/diagnostic/js/adapters-local-change-not-aware.ts

```diff
@@ -34,39 +51,43 @@ export class AdaptersLocalChangeNotAware implements DiagnosticProducer<Node> {
         textDocument: TextDocument,
         node: Node
     ): Promise<Diagnostic[]> {
-        return Promise.resolve(
-            this.findLocalChangeNotAwareAdapterNode(
-                node,
-                LOCAL_CHANGE_NOT_AWARE_ADAPTERS
-            ).map((item) => {
-                return {
-                    severity: SEVERITY,
-                    range: {
-                        start: textDocument.positionAt(item.start as number),
-                        end: textDocument.positionAt(item.end as number)
-                    },
-                    message: LOCAL_CHANGE_NOT_AWARE_MESSAGE
-                } as Diagnostic;
-            })
-        );
+        const result: Diagnostic[] = [];
+
+        const nodesWithAdapter = this.findLocalChangeNotAwareAdapterNode(node);
+
+        for (let [node, info] of nodesWithAdapter) {
+            const diagnostic: Diagnostic = {
+                severity: SEVERITY,
+                range: {
+                    start: textDocument.positionAt(node.start as number),
+                    end: textDocument.positionAt(node.end as number)
+                },
+                message: info.message
+            };
+            if (info.link !== undefined) {
+                diagnostic.code = info.link;
+                diagnostic.codeDescription = {
+                    href: info.link
+                };
+            }
+            result.push(diagnostic);
+        }
+
+        return Promise.resolve(result);
     }
 
     /**
-     * Find @wire adapter call which called in the local change not aware adapters. For example: 
+     * Find a list of adapters which are not draft change aware. For example: getRelatedListRecords from below LWC. 
         export default class RelatedListRecords extends LightningElement {
             ...
             @wire(getRelatedListRecords, 
             ...
         }
-     * @param astNode root node to search
-     * @param adapterNames adapter which are not able to reflect the local change.
-     * @returns nodes with adapter name
+     * @param astNode The AST root node for the LWC js file.
+     * @returns A map of node to adapter name which is not draft aware. 
```

Comment:
How about `A map where each key is an ASTNode representing an adapter that does not account for specific local changes, and each value is an Info object containing a message and, optionally, a link with additional context.`